### PR TITLE
WMS GetFeatureInfo and TIME parameter

### LINKED ISSUE #42141: WMS GetFeatureInfo on time enabled layer does not pass TIME parameter
the same issue as described here - https://github.com/qgis/QGIS/issues/37191
....
**Describe the bug**

Using the new temporal functionality in QGIS 3.18.0, adding a time enabled WMS results in GetFeatureInfo requests not passing TIME accordingly. This results in values on the canvas not corresponding to the values of a given point/pixel when using the identify tool.

**How to Reproduce**

1. Add https://geo.weather.gc.ca/geomet via the WMS provider
2. Enable temporal capability
3. Click the 'identify' tool
4. Show Debugging/Development Tools Panel
5. Query any part of the temporal layer on the map canvas
6. See resulting GetFeatureInfo request in Debugging/Development Tools Panel without TIME= passed

sample request produced by qgis: 
curl 'https://geo.weather.gc.ca/geomet?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetFeatureInfo&BBOX=58.37233146067413259,-168.08258426966281718,64.27858146067413259,-155.77900280898867891&CRS=EPSG:4326&WIDTH=927&HEIGHT=445&LAYERS=RADAR_1KM_RDBR&STYLES=&FORMAT=image/png&QUERY_LAYERS=RADAR_1KM_RDBR&INFO_FORMAT=application/vnd.ogc.gml&I=327&J=302' -H 'User-Agent: Mozilla/5.0 QGIS/31800' --compressed

**QGIS and OS versions**
qgis 3.18.0

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

Fixes https://github.com/qgis/QGIS/issues/42141.

### CODE REVIEW

File: python/core/auto_generated/raster/qgsrasterdataprovidertemporalcapabilities.sip.in [hunk 1]
```diff
@@ -140,6 +140,20 @@ datetime values for the provider.
 .. seealso:: :py:func:`defaultInterval`
 
 .. versionadded:: 3.20
+%End
+
+    void setRequestedTemporalRange( const QgsDateTimeRange &range );
```
Comment (nyalldawson) [inline python/core/auto_generated/raster/qgsrasterdataprovidertemporalcapabilities.sip.in:145]: Can we skip this from the sip bindings? I'm still not 100% on this particular bit of api, and don't want it locked into stable api yet

File: src/gui/qgsmaptoolidentify.cpp [hunk 2]
```diff
@@ -944,7 +944,7 @@ bool QgsMapToolIdentify::identifyRasterLayer( QList<IdentifyResult> *results, Qg
   if ( !layer )
     return false;
 
-  QgsRasterDataProvider *dprovider = layer->dataProvider();
+  QgsRasterDataProvider *dprovider = layer->dataProvider()->clone();
```
Comment (nyalldawson) [inline src/gui/qgsmaptoolidentify.cpp:947]: This leaks

File: tests/src/app/testqgsmaptoolidentifyaction.cpp [hunk 3]
```diff
@@ -565,6 +572,33 @@ TestQgsMapToolIdentifyAction::testIdentifyVectorTile( QgsVectorTileLayer *layer,
...
+  QgsDateTimeRange range = QgsDateTimeRange( QDateTime( QDate( 2020, 1, 1 ), QTime(), Qt::UTC ),
+                           QDateTime( QDate( 2020, 3, 31 ), QTime(), Qt::UTC ) );
+  qobject_cast< QgsRasterLayerTemporalProperties * >( tempLayer->temporalProperties() )->setFixedTemporalRange( range );
+
+  canvas->setExtent( QgsRectangle( 0, 0, 7, 1 ) );
+
+  // invalid temporal range on canvas
+  canvas->setTemporalRange( QgsDateTimeRange( QDateTime( QDate( 1950, 01, 01 ), QTime( 0, 0, 0 ), Qt::UTC ),
+                            QDateTime( QDate( 1950, 01, 01 ), QTime( 1, 0, 0 ), Qt::UTC ) ) );
+  QCOMPARE( testIdentifyRaster( tempLayer.get(), 0.5, 0.5 ), QString( "" ) );
```
Comment (elpaso) [inline tests/src/app/testqgsmaptoolidentifyaction.cpp:594]:
```suggestion
  QCOMPARE( testIdentifyRaster( tempLayer.get(), 0.5, 0.5 ), QString( ) );
```

File: tests/src/app/testqgsmaptoolidentifyaction.cpp [hunk 4]
```diff
@@ -565,6 +572,33 @@ TestQgsMapToolIdentifyAction::testIdentifyVectorTile( QgsVectorTileLayer *layer,
   return result;
 }
 
+void TestQgsMapToolIdentifyAction::identifyRasterTemporal()
+{
+  //create a temporary layer
+  QString raster = QStringLiteral( TEST_DATA_DIR ) + "/raster/test.asc";
+  std::unique_ptr< QgsRasterLayer> tempLayer( new QgsRasterLayer( raster ) );
```
Comment (elpaso) [inline tests/src/app/testqgsmaptoolidentifyaction.cpp:579]: You could use `std::make_unique`

**Review by elpaso — APPROVED** [review]

**Pull request merged**